tx-tracker/consumer/target_processor.go

```diff
@@ -133,27 +138,39 @@ func calculateFeeDetail(params *ProcessTargetTxParams, logger *zap.Logger) *FeeD
 			)
 			return nil
 		}
-		if fee == "" {
-			return nil
-		}
-		return &FeeDetail{
-			RawFee: map[string]string{
-				"gasUsed":           params.EvmFee.GasUsed,
-				"effectiveGasPrice": params.EvmFee.EffectiveGasPrice,
-			},
-			Fee: fee,
+		if fee != "" {
+			feeDetail = &FeeDetail{
+				RawFee: map[string]string{
+					"gasUsed":           params.EvmFee.GasUsed,
+					"effectiveGasPrice": params.EvmFee.EffectiveGasPrice,
+				},
+				Fee: fee,
+			}
 		}
 	}
 	// calculate tx fee for solana redeemed tx.
 	if params.SolanaFee != nil {
 		fee := chains.SolanaCalculateFee(params.SolanaFee.Fee)
-		return &FeeDetail{
+		feeDetail = &FeeDetail{
 			RawFee: map[string]string{
 				"fee": strconv.FormatUint(params.SolanaFee.Fee, 10),
 			},
 			Fee: fee,
 		}
 	}
 
-	return nil
+	if feeDetail != nil && params.P2pNetwork == domain.P2pMainNet {
+		gasPrice, errGasPrice := chains.GetGasPrice(params.ChainID, notionalCache)
+		if errGasPrice != nil {
+			logger.Error("Failed to get gas price",
+				zap.Error(errGasPrice),
+				zap.String("chainId", params.ChainID.String()),
+				zap.String("txHash", params.TxHash),
+			)
+			return feeDetail
+		}
+		feeDetail.RawFee["gasPrice"] = gasPrice.NotionalUsd.String()
```

Comment:
Related to the comment about new fields. Would it make sense to do something like this?

```suggestion
		feeDetail.GasTokenNotional = gasPrice.NotionalUsd.String()
		fee, err := decimals.NewFromString(feeDetail.Fee)
		if err != nil {
			logger.Error("Failed to convert fee to decimal",
				zap.Error(err),
				zap.String("chainId", params.ChainID.String()),
				zap.String("txHash", params.TxHash),
			)
		}
		feeDetail.FeeUsd = fee.Mul(gasPrice.NotionalUsd)		
```